### PR TITLE
GitHub Follower/ing On Dashboard

### LINKED ISSUE #14: User Dashboard: Github Following
👇 👇 👇 👇 
**REMINDER:** Don't start this story until you and your partner pair on and deploy the previous user story.
☝️ ☝️ ☝️ ☝️ 

**Details:** A user exists in the database. They have a Github token stored. Generate a token here for testing and development: https://github.com/settings/tokens

A future story will implement generating these tokens dynamically but that isn't necessary here.

```
As a logged in user
When I visit /dashboard
Then I should see a section for "Github"
And under that section I should see another section titled "Following"
And I should see list of users I follow with their handles linking to their Github profile
```

**API endpoint documentation**
https://developer.github.com/v3/users/followers/#list-users-followed-by-another-user

- [ ] API call uses authenticated call
- [ ] No hashes in the view.
- [ ] Uses objects in the view to represent the Github users.

### LINKED ISSUE #15: User Dashboard: Github Followers
🚔 👮‍♀️ 👮‍♀️ 👮‍♀️ 
**DO NOT START THIS STORY UNLESS THE STORY REFERENCED BELOW HAS BEEN COMPLETED BY BOTH IN YOUR PAIR.**

You should pair on story  "User Dashboard: Github Repositories" and deploy it prior to working on this one.

Continue this process for all stories labeled `spike first`. You will learn more this way and produce better code.

🚓 💨 👮 👮 👮 

**Details:** A user exists in the database. They have a Github token stored. Generate a token here for testing and development: https://github.com/settings/tokens

A future story will implement generating these tokens dynamically but that isn't necessary here.

No need to hit the endpoint for followings. A future story will address this.

```
As a logged in user
When I visit /dashboard
Then I should see a section for "Github"
And under that section I should see another section titled "Followers"
And I should see list of all followers with their handles linking to their Github profile
```

**API endpoint documentation**
https://developer.github.com/v3/users/followers/

- [ ] API call uses authenticated call: `/user/followers` instead of `/users/:username/followers`
- [ ] No hashes in the view.
- [ ] Uses objects in the view to represent the followers.

### PR DESCRIPTION
**Pull request by wipegup**
Closes #14 
Closes #15 

Adds Additional Github information - of all followers and of all users followed by user,  to user dashboard (when GitHub Token is present)

### CODE REVIEW

File: spec/features/user/user_sees_github_info_on_dashboard_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+describe 'A registered user', :vcr do
```
[the remark this answers is not in the document]
Comment (wipegup) [reply spec/features/user/user_sees_github_info_on_dashboard_spec.rb]: Longer Blocks expected in Tests

File: app/services/git_hub_service.rb [hunk 2]
```diff
@@ -9,6 +9,16 @@ def get_repos
...
 
+  def retrieve_followers
+    response = conn.get('followers')
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def retrieve_following
+    response = conn.get('following')
+    JSON.parse(response.body, symbolize_names: true)
+  end
```
Comment (milevy1) [inline app/services/git_hub_service.rb]: The 3 above methods: get_repos, retrieve_followers, and retreive_following are all very similar.  To DRY this up, what do you think about refactoring to one method that took one argument, a string, of what you are retrieving? ("repos", "followers", "following")

File: app/services/git_hub_service.rb [hunk 3]
```diff
@@ -4,11 +4,11 @@ def initialize(token)
...
   end
 
-  def get_repos
-    response = conn.get('repos')
+  def retrieve(url_end)
+    response = conn.get(url_end)
     JSON.parse(response.body, symbolize_names: true)
   end
-
+
```
[the remark this answers is not in the document]
Comment (wipegup) [reply app/services/git_hub_service.rb]: Single line wanted between funcitons

File: app/services/git_hub_service.rb (hunk 2 above)
Comment (wipegup) [reply app/services/git_hub_service.rb]: Addressed in latest commit

**Pull request merged**